### PR TITLE
feat(securitygroups): Add option to delete default outbound rule

### PR DESCRIPTION
**Pull request by gvdhart**
Add option in security group spec to remove default outbound rule that allows all outbound traffic

### CODE REVIEW

File: config/crd/bases/infrastructure.cluster.x-k8s.io_oscclusters.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.15.0
+    controller-gen.kubebuilder.io/version: v0.14.0
```
Comment (outscale-hmi) [inline config/crd/bases/infrastructure.cluster.x-k8s.io_oscclusters.yaml:6]: why did you dowgrade this version ?

File: controllers/osccluster_securitygroup_controller.go [hunk 2]
```diff
@@ -342,6 +343,14 @@ func reconcileSecurityGroup(ctx context.Context, clusterScope *scope.ClusterScop
 				}
 			}
 		}
+
+		if deleteDefaultOutboundRule {
```
Comment (outscale-hmi) [inline controllers/osccluster_securitygroup_controller.go:347]: Can you add a unit test ?

**Review by outscale-hmi — CHANGES_REQUESTED** [review]

File: config/crd/bases/infrastructure.cluster.x-k8s.io_oscclusters.yaml (hunk 1 above)
Comment (gvdhart) [reply config/crd/bases/infrastructure.cluster.x-k8s.io_oscclusters.yaml:6]: Didn't do it on purpose, I am not up to date on my controller-gen version, will change that.

**Comment by gvdhart** [comment]
Will add this here instead: https://github.com/outscale/cluster-api-provider-outscale/pull/373

**Comment by gvdhart** [comment]
Will add this here instead: https://github.com/outscale/cluster-api-provider-outscale/pull/373

**Pull request closed without merging**